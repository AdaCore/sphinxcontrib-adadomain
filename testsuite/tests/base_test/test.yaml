driver: gen-doc
project_file: prj/prj.gpr
