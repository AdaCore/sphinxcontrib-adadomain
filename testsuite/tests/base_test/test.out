### pkg.xml:

<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE document PUBLIC "+//IDN docutils.sourceforge.net//DTD Docutils Generic//EN//XML" "http://docutils.sourceforge.net/docs/ref/docutils.dtd">
<!-- Generated by Docutils 0.19 -->
    <section ids="package-Pkg Pkg pkg" names="pkg">
        <title>Pkg</title>
        <index entries="['single',\ 'Pkg\ (package)',\ 'package-Pkg',\ 'Pkg',\ None]"></index>
        <paragraph>Top-level package documentation.  I’m baby la croix drinking vinegar
            actually, photo booth pinterest raw denim coloring book occupy meggings
            church-key yr. Waistcoat tofu bruh mustache cornhole butcher normcore
            forage Brooklyn ramps. Sus PBR&amp;B, cupping VHS swag tofu poutine authentic
            godard actually chia.</paragraph>
        <section ids="header" names="header">
            <title>Header</title>
            <paragraph>Shoreditch you probably haven’t heard of them four loko yr, pour-over
                typewriter try-hard beard. Roof party live-edge jean shorts tilde iPhone,
                everyday carry small batch knausgaard disrupt solarpunk tacos. Ascot yes
                plz live-edge ramps narwhal heirloom pok pok. Pour-over kombucha
                intelligentsia, salvia health goth gatekeep butcher wayfarers lo-fi
                succulents.</paragraph>
            <bullet_list bullet="*">
                <list_item>
                    <paragraph>List</paragraph>
                </list_item>
                <list_item>
                    <paragraph>Of elements</paragraph>
                </list_item>
            </bullet_list>
            <paragraph>Stumptown cornhole you probably haven’t heard of them ramps, try-hard pork
                belly bodega boys bushwick meditation grailed keytar gorpcore marxism
                portland lo-fi.</paragraph>
            <index entries="['single',\ 'Pkg.T\ (Ada\ type)',\ 'Pkg.T',\ '',\ None]"></index>
            <desc classes="ada type" desctype="type" domain="ada" nocontentsentry="False" noindex="False" noindexentry="False" objtype="type">
                <desc_signature _toc_name="" _toc_parts="" classes="sig sig-object ada sig sig-object" fullname="Pkg.T" ids="Pkg.T" package="Pkg"><desc_annotation xml:space="preserve">type </desc_annotation><desc_name classes="sig-name descname sig-name descname" xml:space="preserve">T</desc_name><desc_type xml:space="preserve"></desc_type></desc_signature>
                <desc_content>
                    <field_list>
                        <field>
                            <field_name>Components</field_name>
                            <field_body>
                                <bullet_list>
                                    <list_item>
                                        <paragraph><literal_strong>A</literal_strong> (<inline><literal classes="xref ada ada-type">Standard.Integer</literal></inline>) – Documentation for A, B, C</paragraph>
                                    </list_item>
                                    <list_item>
                                        <paragraph><literal_strong>B</literal_strong> (<inline><literal classes="xref ada ada-type">Standard.Integer</literal></inline>) – Documentation for A, B, C</paragraph>
                                    </list_item>
                                    <list_item>
                                        <paragraph><literal_strong>C</literal_strong> (<inline><literal classes="xref ada ada-type">Standard.Integer</literal></inline>) – Documentation for A, B, C</paragraph>
                                    </list_item>
                                </bullet_list>
                            </field_body>
                        </field>
                    </field_list>
                    <paragraph>Documentation for A, B, C</paragraph>
                    <index entries="['single',\ 'Pkg.Foo\ (Ada\ procedure)',\ 'Pkg.Foo',\ '',\ None]"></index>
                    <desc classes="ada procedure" desctype="procedure" domain="ada" nocontentsentry="False" noindex="False" noindexentry="False" objtype="procedure">
                        <desc_signature _toc_name="" _toc_parts="" classes="sig sig-object ada sig sig-object" fullname="Pkg.Foo" ids="Pkg.Foo" package="Pkg"><desc_annotation xml:space="preserve">procedure </desc_annotation><desc_name classes="sig-name descname sig-name descname" xml:space="preserve">Foo</desc_name> <desc_parameterlist xml:space="preserve"><desc_parameter xml:space="preserve"><desc_sig_name classes="n n">Self</desc_sig_name><desc_sig_punctuation classes="p p"> : </desc_sig_punctuation><reference internal="True" refid="Pkg.T" reftitle="Pkg.T"><desc_sig_name classes="n n n">T</desc_sig_name></reference></desc_parameter></desc_parameterlist></desc_signature>
                        <desc_content>
                        </desc_content>
                    </desc>
                    <index entries="['single',\ 'Pkg.Create\ (Ada\ function)',\ 'Pkg.Create',\ '',\ None]"></index>
                    <desc classes="ada function" desctype="function" domain="ada" nocontentsentry="False" noindex="False" noindexentry="False" objtype="function">
                        <desc_signature _toc_name="" _toc_parts="" classes="sig sig-object ada sig sig-object" fullname="Pkg.Create" ids="Pkg.Create" package="Pkg"><desc_annotation xml:space="preserve">function </desc_annotation><desc_name classes="sig-name descname sig-name descname" xml:space="preserve">Create</desc_name> <desc_parameterlist xml:space="preserve"><desc_parameter xml:space="preserve"><desc_sig_name classes="n n">A</desc_sig_name><desc_sig_punctuation classes="p p">, </desc_sig_punctuation><desc_sig_name classes="n n">B</desc_sig_name><desc_sig_punctuation classes="p p">, </desc_sig_punctuation><desc_sig_name classes="n n">C</desc_sig_name><desc_sig_punctuation classes="p p"> : </desc_sig_punctuation><desc_sig_name classes="n n n">Standard.Integer</desc_sig_name></desc_parameter></desc_parameterlist><reference internal="True" refid="Pkg.T" reftitle="Pkg.T"><desc_returns xml:space="preserve">T</desc_returns></reference></desc_signature>
                        <desc_content>
                            <paragraph>Constructor function for <reference internal="True" refid="Pkg.T" reftitle="Pkg.T"><literal classes="xref ada ada-ref">T</literal></reference></paragraph>
                        </desc_content>
                    </desc>
                    <index entries="['single',\ 'Pkg.Poo\ (Ada\ function)',\ 'Pkg.Poo',\ '',\ None]"></index>
                    <desc classes="ada function" desctype="function" domain="ada" nocontentsentry="False" noindex="False" noindexentry="False" objtype="function">
                        <desc_signature _toc_name="" _toc_parts="" classes="sig sig-object ada sig sig-object" fullname="Pkg.Poo" ids="Pkg.Poo" package="Pkg"><desc_annotation xml:space="preserve">function </desc_annotation><desc_name classes="sig-name descname sig-name descname" xml:space="preserve">Poo</desc_name> <desc_parameterlist xml:space="preserve"><desc_parameter xml:space="preserve"><desc_sig_name classes="n n">Self</desc_sig_name><desc_sig_punctuation classes="p p"> : </desc_sig_punctuation><reference internal="True" refid="Pkg.T" reftitle="Pkg.T"><desc_sig_name classes="n n n">T</desc_sig_name></reference></desc_parameter></desc_parameterlist><desc_returns xml:space="preserve">Standard.Boolean</desc_returns></desc_signature>
                        <desc_content>
                            <paragraph>Check whether the <literal>@</literal> shortcut syntax is handled correctly by
                                referencing <literal classes="xref ada ada-ref">U.D</literal></paragraph>
                        </desc_content>
                    </desc>
                    <index entries=""></index>
                    <desc classes="ada object" desctype="object" domain="ada" nocontentsentry="False" noindex="False" noindexentry="False" objtype="object">
                        <desc_signature _toc_name="" _toc_parts="" classes="sig sig-object ada sig sig-object" fullname="Pkg.Singleton" ids="Pkg.Singleton" package="Pkg"><desc_name classes="sig-name descname sig-name descname" xml:space="preserve">Singleton</desc_name><desc_annotation xml:space="preserve"> : T</desc_annotation><desc_type xml:space="preserve"></desc_type></desc_signature>
                        <desc_content>
                            <field_list>
                                <field>
                                    <field_name>Object type</field_name>
                                    <field_body>
                                        <paragraph><inline><reference internal="True" refid="Pkg.T" reftitle="Pkg.T"><literal classes="xref ada ada-type">T</literal></reference></inline></paragraph>
                                    </field_body>
                                </field>
                                <field>
                                    <field_name>Default value</field_name>
                                    <field_body>
                                        <paragraph><literal>(1, 2, 3)</literal></paragraph>
                                    </field_body>
                                </field>
                            </field_list>
                            <paragraph>Documentation for Singleton</paragraph>
                        </desc_content>
                    </desc>
                    <index entries=""></index>
                    <desc classes="ada object" desctype="object" domain="ada" nocontentsentry="False" noindex="False" noindexentry="False" objtype="object">
                        <desc_signature _toc_name="" _toc_parts="" classes="sig sig-object ada sig sig-object" fullname="Pkg.Singleton2" ids="Pkg.Singleton2" package="Pkg"><desc_name classes="sig-name descname sig-name descname" xml:space="preserve">Singleton2</desc_name><desc_annotation xml:space="preserve"> : T</desc_annotation><desc_type xml:space="preserve"></desc_type></desc_signature>
                        <desc_content>
                            <field_list>
                                <field>
                                    <field_name>Object type</field_name>
                                    <field_body>
                                        <paragraph><inline><reference internal="True" refid="Pkg.T" reftitle="Pkg.T"><literal classes="xref ada ada-type">T</literal></reference></inline></paragraph>
                                    </field_body>
                                </field>
                            </field_list>
                            <paragraph>Documentation for Singleton 2</paragraph>
                        </desc_content>
                    </desc>
                </desc_content>
            </desc>
            <index entries="['single',\ 'Pkg.U\ (Ada\ type)',\ 'Pkg.U',\ '',\ None]"></index>
            <desc classes="ada type" desctype="type" domain="ada" nocontentsentry="False" noindex="False" noindexentry="False" objtype="type">
                <desc_signature _toc_name="" _toc_parts="" classes="sig sig-object ada sig sig-object" fullname="Pkg.U" ids="Pkg.U" package="Pkg"><desc_annotation xml:space="preserve">type </desc_annotation><desc_name classes="sig-name descname sig-name descname" xml:space="preserve">U</desc_name><desc_type xml:space="preserve"></desc_type></desc_signature>
                <desc_content>
                    <field_list>
                        <field>
                            <field_name>Components</field_name>
                            <field_body>
                                <bullet_list>
                                    <list_item>
                                        <paragraph><literal_strong>C</literal_strong> (<inline><literal classes="xref ada ada-type">Standard.Float</literal></inline>) – </paragraph>
                                    </list_item>
                                    <list_item>
                                        <paragraph><literal_strong>D</literal_strong> (<inline><literal classes="xref ada ada-type">Standard.Float</literal></inline>) – </paragraph>
                                    </list_item>
                                    <list_item>
                                        <paragraph><literal_strong>E</literal_strong> (<inline><literal classes="xref ada ada-type">Standard.Float</literal></inline>) – </paragraph>
                                    </list_item>
                                </bullet_list>
                            </field_body>
                        </field>
                    </field_list>
                    <index entries="['single',\ 'Pkg.Primitive_Of_Both\ (Ada\ procedure)',\ 'Pkg.Primitive_Of_Both',\ '',\ None]"></index>
                    <desc classes="ada procedure" desctype="procedure" domain="ada" nocontentsentry="False" noindex="False" noindexentry="False" objtype="procedure">
                        <desc_signature _toc_name="" _toc_parts="" classes="sig sig-object ada sig sig-object" fullname="Pkg.Primitive_Of_Both" ids="Pkg.Primitive_Of_Both" package="Pkg"><desc_annotation xml:space="preserve">procedure </desc_annotation><desc_name classes="sig-name descname sig-name descname" xml:space="preserve">Primitive_Of_Both</desc_name> <desc_parameterlist xml:space="preserve"><desc_parameter xml:space="preserve"><desc_sig_name classes="n n">Self</desc_sig_name><desc_sig_punctuation classes="p p"> : </desc_sig_punctuation><reference internal="True" refid="Pkg.T" reftitle="Pkg.T"><desc_sig_name classes="n n n">T</desc_sig_name></reference></desc_parameter><desc_parameter xml:space="preserve"><desc_sig_name classes="n n">Other</desc_sig_name><desc_sig_punctuation classes="p p"> : </desc_sig_punctuation><reference internal="True" refid="Pkg.U" reftitle="Pkg.U"><desc_sig_name classes="n n n">U</desc_sig_name></reference></desc_parameter></desc_parameterlist></desc_signature>
                        <desc_content>
                            <paragraph>This is a primitive of both types. We want to test that thanks to the
                                <literal>belongs-to</literal> annotation, it is correctly attached to <reference internal="True" refid="Pkg.U" reftitle="Pkg.U"><literal classes="xref ada ada-ref">U</literal></reference>.</paragraph>
                        </desc_content>
                    </desc>
                </desc_content>
            </desc>
            <index entries=""></index>
            <desc classes="ada generic_package" desctype="generic_package" domain="ada" nocontentsentry="False" noindex="False" noindexentry="False" objtype="generic_package">
                <desc_signature _toc_name="" _toc_parts="" classes="sig sig-object ada sig sig-object" fullname="Pkg.Pkg.Gen_Package" ids="Pkg.Pkg.Gen_Package" package="Pkg"><desc_annotation xml:space="preserve">generic package </desc_annotation><desc_name classes="sig-name descname sig-name descname" xml:space="preserve">Pkg.Gen_Package</desc_name></desc_signature>
                <desc_content>
                    <paragraph>Documentation for Singleton 2</paragraph>
                    <field_list>
                        <field>
                            <field_name>Formals</field_name>
                            <field_body>
                                <index entries="['single',\ 'Pkg.Gen_Package.F\ (Ada\ type)',\ 'Pkg.Gen_Package.F',\ '',\ None]"></index>
                                <desc classes="ada type" desctype="type" domain="ada" nocontentsentry="False" noindex="False" noindexentry="False" objtype="type">
                                    <desc_signature _toc_name="" _toc_parts="" classes="sig sig-object ada sig sig-object" fullname="Pkg.Gen_Package.F" ids="Pkg.Gen_Package.F" package="Pkg.Gen_Package"><desc_annotation xml:space="preserve">type </desc_annotation><desc_name classes="sig-name descname sig-name descname" xml:space="preserve">F</desc_name><desc_type xml:space="preserve"></desc_type></desc_signature>
                                    <desc_content>
                                        <paragraph>The main type for this generic package</paragraph>
                                    </desc_content>
                                </desc>
                                <index entries="['single',\ 'Pkg.Gen_Package.Frobulize\ (Ada\ function)',\ 'Pkg.Gen_Package.Frobulize',\ '',\ None]"></index>
                                <desc classes="ada function" desctype="function" domain="ada" nocontentsentry="False" noindex="False" noindexentry="False" objtype="function">
                                    <desc_signature _toc_name="" _toc_parts="" classes="sig sig-object ada sig sig-object" fullname="Pkg.Gen_Package.Frobulize" ids="Pkg.Gen_Package.Frobulize" package="Pkg.Gen_Package"><desc_annotation xml:space="preserve">function </desc_annotation><desc_name classes="sig-name descname sig-name descname" xml:space="preserve">Frobulize</desc_name> <desc_parameterlist xml:space="preserve"><desc_parameter xml:space="preserve"><desc_sig_name classes="n n">Self</desc_sig_name><desc_sig_punctuation classes="p p"> : </desc_sig_punctuation><reference internal="True" refid="Pkg.Gen_Package.F" reftitle="Pkg.Gen_Package.F"><desc_sig_name classes="n n n">F</desc_sig_name></reference></desc_parameter></desc_parameterlist><desc_returns xml:space="preserve">Standard.Boolean</desc_returns></desc_signature>
                                    <desc_content>
                                        <paragraph>A way to frobulize instances</paragraph>
                                    </desc_content>
                                </desc>
                            </field_body>
                        </field>
                    </field_list>
                </desc_content>
            </desc>
            <index entries=""></index>
            <desc classes="ada exception" desctype="exception" domain="ada" nocontentsentry="False" noindex="False" noindexentry="False" objtype="exception">
                <desc_signature _toc_name="" _toc_parts="" classes="sig sig-object ada sig sig-object" fullname="Pkg.Froob" ids="Pkg.Froob" package="Pkg"><desc_name classes="sig-name descname sig-name descname" xml:space="preserve">Froob</desc_name><desc_annotation xml:space="preserve">: exception</desc_annotation></desc_signature>
                <desc_content>
                    <paragraph>A custom exception, raised in the implementation of <reference internal="True" refid="Pkg.U" reftitle="Pkg.U"><literal classes="xref ada ada-ref">U</literal></reference></paragraph>
                </desc_content>
            </desc>
            <index entries=""></index>
            <desc classes="ada package" desctype="package" domain="ada" nocontentsentry="False" noindex="False" noindexentry="False" objtype="package">
                <desc_signature _toc_name="" _toc_parts="" classes="sig sig-object ada sig sig-object" fullname="Pkg.Pkg.Nested_Package" ids="Pkg.Pkg.Nested_Package" package="Pkg"><desc_annotation xml:space="preserve">package </desc_annotation><desc_name classes="sig-name descname sig-name descname" xml:space="preserve">Pkg.Nested_Package</desc_name></desc_signature>
                <desc_content>
                    <paragraph>This is a nested package</paragraph>
                    <index entries="['single',\ 'Pkg.Nested_Package.Barize\ (Ada\ function)',\ 'Pkg.Nested_Package.Barize',\ '',\ None]"></index>
                    <desc classes="ada function" desctype="function" domain="ada" nocontentsentry="False" noindex="False" noindexentry="False" objtype="function">
                        <desc_signature _toc_name="" _toc_parts="" classes="sig sig-object ada sig sig-object" fullname="Pkg.Nested_Package.Barize" ids="Pkg.Nested_Package.Barize" package="Pkg.Nested_Package"><desc_annotation xml:space="preserve">function </desc_annotation><desc_name classes="sig-name descname sig-name descname" xml:space="preserve">Barize</desc_name> <desc_parameterlist xml:space="preserve"><desc_parameter xml:space="preserve"><desc_sig_name classes="n n">Inst</desc_sig_name><desc_sig_punctuation classes="p p"> : </desc_sig_punctuation><reference internal="True" refid="Pkg.T" reftitle="Pkg.T"><desc_sig_name classes="n n n">T</desc_sig_name></reference></desc_parameter><desc_parameter xml:space="preserve"><desc_sig_name classes="n n">Other_Inst</desc_sig_name><desc_sig_punctuation classes="p p"> : </desc_sig_punctuation><reference internal="True" refid="Pkg.U" reftitle="Pkg.U"><desc_sig_name classes="n n n">U</desc_sig_name></reference></desc_parameter></desc_parameterlist><desc_returns xml:space="preserve">Standard.Boolean</desc_returns></desc_signature>
                        <desc_content>
                            <paragraph>Barize the items</paragraph>
                        </desc_content>
                    </desc>
                </desc_content>
            </desc>
        </section>
    </section>
</document>

